Source/WebCore/page/SecurityOrigin.cpp

```diff
@@ -176,7 +128,14 @@ bool shouldTreatAsPotentiallyTrustworthy(const URL& url)
 }
 
 SecurityOrigin::SecurityOrigin(const URL& url)
-    : m_data(SecurityOriginData::fromURL(url))
+    : SecurityOrigin(SecurityOriginData::fromURL(url))
+{
+    if (m_canLoadLocalResources)
+        m_filePath = url.fileSystemPath(); // In case enforceFilePathSeparation() is called.
+}
+
+SecurityOrigin::SecurityOrigin(SecurityOriginData&& data)
```

Comment:
It seems `m_canLoadLocalResources` might not be set for a SecurityOrigin object created by `SecurityOrigin::create(SecurityOriginData&& data)` because `SecurityOrigin::SecurityOrigin(SecurityOriginData&& data)` does not set it any more.

Reply:
I'm assuming you meant `m_filePath`, instead of `m_canLoadLocalResources` (`m_canLoadLocalResources` is set by `m_canLoadLocalResources = isLocal();`).

I believe you're correct that we should set `m_filePath` in `SecurityOrigin::SecurityOrigin(SecurityOriginData&&)`, too. I'll fix that.